src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -767,6 +774,67 @@ private void assertNotNull(AssertionInfo info, CharSequence actual) {
     Objects.instance().assertNotNull(info, actual);
   }
 
+  /**
+   * Verifies that the given charSequence contains exactly the given charSequences in a sequential order with no more or less elements.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given charSequence.
+   * @param values the charSequences to check.
+   * @throws NullPointerException if the array of values is {@code null}.
+   * @throws IllegalArgumentException if the array of values is empty.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the given {@code CharSequence} does not contain exactly the values in a sequential order.
+   */
+  public void assertContainsExactly(AssertionInfo info, CharSequence actual, CharSequence[] values) {
+    doCommonCheckForCharSequence(info, actual, values);
+
+    List<String> actualAsList = splitCharSequencesToListOfElements(actual);
+    List<String> valuesAsList = splitCharSequencesToListOfElements(values);
+
+    if (actualAsList.size() != valuesAsList.size()) {
+      throw failures.failure(info, shouldHaveSameSizeAs(actual, actualAsList.size(), valuesAsList.size()));
+    }
+
+    IterableDiff diff = diff(actualAsList, valuesAsList, comparisonStrategy);
+
+    if (!diff.differencesFound()) {
+      checkIdenticalElementsAreInSameOrder(info, actualAsList, valuesAsList);
+      return;
+    }
+
```

Comment:
Although the algorithm is correct, the error reported with `shouldContainExactly` is not user friendly as it shows an array of the missing/extra characters instead of missing/extra strings. 
By splitting strings into chars, we loose how users have grouped chars into strings parameters and not a collection of chars.

I would rather show the diff with the missing/extra strings instead, for example
```
assertThat("Practice makes perfect!").containsExactly("Practice ", "makes ", "!");
```
should show something lie
```
java.lang.AssertionError: 
Expected: <java.lang.AssertionError: 
Expecting:
  <"Practice makes perfect!">
to contain exactly in any order:
  <["Practice ", "makes ", "!"]>
but the following elements were not found:
  <["perfect"]>
>
```
I haven't thought of a nice way of doing that but let's give it a try.